scss/_dropdown.scss

```diff
@@ -32,19 +32,29 @@
   border: $dropdown-border-width solid $dropdown-border-color;
   @include border-radius($dropdown-border-radius);
   @include box-shadow($dropdown-box-shadow);
+
+  // Reset positioning when positioned with Popper
+  &[style] {
```

Comment:
Isn't this a bit too generic? I mean, what if someone has an inline `style` in their `.dropdown-menu`?

Reply:
I know, but there is just no other way. Tried the `data-popper-placement` attribute selector, but that didn't seem to work.

Remember when I said positioning things with popper was I so difficult? Well, this is one of the things what made it so hard. Spend at least 10 hours to find a solution and this was the only thing that worked.

Reply:
I know, but this PR feels too hacky :/ And as much as I'd like us to move to popper 2.x, unless we find a robust way, this will probably bit us in the a$$ later.

Reply:
`data-popper-placement` will be set when Popper positioned an element. I don't see why it shouldn't work.

Reply:
We get this if we change that selector:
![image](https://user-images.githubusercontent.com/11559216/100730090-e2b6b580-33c9-11eb-8a1e-8126b2aaa067.png)

Reply:
You should check what's happening at CSS level, the selector is working.

Reply:
We get this if we replace that selector:
![image](https://user-images.githubusercontent.com/11559216/100730464-5e186700-33ca-11eb-97ca-cdc70875092c.png)

And in the HTML we have the `bottom-end` placement:
![image](https://user-images.githubusercontent.com/11559216/100730488-67093880-33ca-11eb-9784-d8a83afdb7da.png)

Reply:
It seems to be a CSS specificity issue.

![image](https://user-images.githubusercontent.com/5382443/100731126-45f51780-33cb-11eb-823b-77a1ce1a7440.png)

I would change it to be:

```css
right: auto !important;
left: initial !important;
```

Reply:
This will override the positioning of popper 😛

Reply:
Isn't that what you are trying to do?

Reply:
Nope, because this way we can't align the dropdown to the right side.

Ideally, what we need would be a placement type `none`, so we can handle the positioning ourselves and popper won't do any positioning with `left/top/right/bottom` properties. Popper would then only reposition using transforms if the element would be offscreen.

Or we keep it as is and use the `style` attribute in our CSS.

Reply:
I'm sorry. I don't think I understand what you are trying to achieve...

Reply:
> I'm sorry. I don't think I understand what you are trying to achieve... 

Yeah, it is quite complicated to explain and understand, unless you take a deep dive into our codebase and get all the edge cases.

> Why can't you let Popper handle everything?

Because:
- Our dropdowns also need to work without popper (eg, see the version dropdown in our header https://v5.getbootstrap.com/docs/5.0/getting-started/introduction/)
- We have responsive alignment classes (https://v5.getbootstrap.com/docs/5.0/components/dropdowns/#responsive-alignment). So the alignment of an element can change depending on the viewport width.

Reply:
1. In that case the selector should be `:not([data-popper-placement])` so that you can style the dropdown if Popper doesn't style it;
2. That's something that should be handled on the JS side of things, listen for the resize event, test media query, check which classes are applied, and call [`popperInstance.setOptions`](https://popper.js.org/docs/v2/constructors/#instance)

Reply:
> In that case the selector should be :not([data-popper-placement]) so that you can style the dropdown if Popper doesn't style it;

Tried that, but popper seems to set this attribute after it positions the element, which leads to a wrong positioning. This is not a Popper bug, but does conflict with the way we hack our own positioning with CSS.

> That's something that should be handled on the JS side of things, listen for the resize event, test media query, check which classes are applied, and call popperInstance.setOptions

The media queries can be adjusted/renamed/added/removed in CSS, so there's no real mapping between the CSS and JS side.


We'll stick with the `style` hack for now, since that seems to work. Might review this later on.